core/src/common.cpp

```diff
@@ -339,33 +339,15 @@ namespace irods::http
 
 	auto map_json_to_user(const nlohmann::json& _json) -> std::optional<std::string>
 	{
-		const auto& oidc_config{irods::http::globals::oidc_configuration()};
-		const static auto user_claim{oidc_config.find("irods_user_claim")};
-		const static auto attribute_mapping{oidc_config.find("user_attribute_mapping")};
+		auto json_res_string{to_string(_json)};
+		auto match_func{irods::http::globals::user_map_lib().get<char* (const char*)>("user_mapper_match")};
+		auto* res{match_func(json_res_string.c_str())};
 
-		if (user_claim != std::end(oidc_config)) {
-			if (auto claim{_json.find(user_claim->get_ref<const std::string&>())}; claim != std::end(_json)) {
-				return {*claim};
-			}
-		}
-		else if (attribute_mapping != std::end(oidc_config)) {
-			const auto& mappings{*attribute_mapping};
-
-			for (auto& [key, value] : mappings.items()) {
-				const auto comparison_func{[&_json](const auto& _iter) -> bool {
-					const auto value_of_interest{_json.find(_iter.key())};
-					if (value_of_interest == std::end(_json)) {
-						return false;
-					}
+		if (nullptr != res) {
+			std::string matched_user{res};
+			std::free(res);
```

Comment:
Might be too much, but should this `free()` call be a `user_mapper_free()` which has its own free implementation? (In this case, just a call to `std::free()`?)

Reply:
Done.